mllib/src/main/scala/org/apache/spark/ml/feature/RFormula.scala

```diff
@@ -200,6 +200,13 @@ class RFormulaModel private[feature](
       dataset.schema(labelName).dataType match {
         case _: NumericType | BooleanType =>
           dataset.withColumn($(labelCol), dataset(labelName).cast(DoubleType))
+        case _: StringType => {
+          val indexer = new StringIndexer()
```

Comment:
This should not happen inside `transform`. we cannot afford one `fit` call per `transform`. We should inside the label indexer into the `RFormula.fit`instead of `RFormulaModel.transform`.